qase-playwright/src/reporter.ts

```diff
@@ -34,48 +44,104 @@ export class PlaywrightQaseReporter implements Reporter {
   };
 
   /**
-   * @type {RegExp}
-   */
-  static qaseIdRegExp = /\(Qase ID: ([\d,]+)\)/;
-
-  /**
-   * @param {string} title
-   * @returns {number[]}
+   * @type {Map<string, number[]>}
    * @private
    */
-  private static getCaseIds(title: string): number[] {
-    const [, ids] = title.match(PlaywrightQaseReporter.qaseIdRegExp) ?? [];
-
-    return ids ? ids.split(',').map((id) => Number(id)) : [];
-  }
+  private static qaseIds: Map<string, number[]> = new Map<string, number[]>();
 
   // private static transformSuiteTitle(test: TestCase) {
   //   return test.titlePath().filter(Boolean);
   // }
 
+  /**
+   * @type {Map<TestStep, TestCase>}
+   * @private
+   */
+  private stepCache: Map<TestStep, TestCase> = new Map<TestStep, TestCase>();
+
+  /**
+   * @type {Map<string, TestStep>}
+   * @private
+   */
+  private stepAttachments: Map<TestStep, Attachment[]> = new Map<TestStep, Attachment[]>();
+
   /**
    * @param {ArrayItemType<TestResult['attachments']>[]} testAttachments
-   * @returns {string[]}
+   * @returns {TestCaseMeteData}
    * @private
    */
-  private static transformAttachments(
+  private transformAttachments(
     testAttachments: ArrayItemType<TestResult['attachments']>[],
-  ): Attachment[] {
+  ): TestCaseMeteData {
+    const meteData: TestCaseMeteData = {
```

Comment:
```suggestion
  ): TestCaseMetaData {
    const metaData: TestCaseMetaData = {
```

Reply:
Maybe even `TestCaseMetadata` and `metadata`, because it's officially a single word https://www.merriam-webster.com/dictionary/metadata

Reply:
fixed